internal/unionstore/art/art_iterator.go

```diff
@@ -50,7 +50,10 @@ func (t *ART) iter(lowerBound, upperBound []byte, reverse, includeFlags bool) (*
 		inner: &baseIter{
 			allocator: &t.allocator,
 		},
-		currAddr: arena.BadAddr, // the default value of currAddr is not equal to any valid address
+		// the default value of currAddr is not equal to any valid address
+		// arena.BadAddr's idx is maxuint32 - 1, which is impossible in common cases,
```

Comment:
The `currAddr` will be set to valid or null address when first time call `Next`, so the only purpose of `BadAddr` is to avoid `currAddr == endAddr` at the beginning.

`BadAddr` can never appear in valid usages, because the [max size](https://github.com/tikv/client-go/blob/527f80a18663674d8d86bfcd7bf7d0d249853ac8/internal/unionstore/arena/arena.go#L49) of block is set, so the offset `maxuint32` is impossible.